File: frontend/lib/models/order.dart
import 'order_item.dart';
import 'user.dart';

class Order {
  final int id;
  final String orderNumber;
  final int userId;
  final double totalAmount;
  final double shippingCost;
  final String status;
  final String paymentStatus;
  final String shippingAddress;
  final String phone;
  final String? transactionId;
  final List<OrderItem> items;
  final DateTime createdAt;
  final DateTime updatedAt;
  final User? user;

  Order({
    required this.id,
    required this.orderNumber,
    required this.userId,
    required this.totalAmount,
    required this.shippingCost,
    required this.status,
    required this.paymentStatus,
    required this.shippingAddress,
    required this.phone,
    this.transactionId,
    required this.items,
    required this.createdAt,
    required this.updatedAt,
    this.user,
  });

  Order copyWith({
    int? id,
    String? orderNumber,
    int? userId,
    double? totalAmount,
    double? shippingCost,
    String? status,
    String? paymentStatus,
    String? shippingAddress,
    String? phone,
    String? transactionId,
    List<OrderItem>? items,
    DateTime? createdAt,
    DateTime? updatedAt,
    User? user,
  }) {
    return Order(
      id: id ?? this.id,
      orderNumber: orderNumber ?? this.orderNumber,
      userId: userId ?? this.userId,
      totalAmount: totalAmount ?? this.totalAmount,
      shippingCost: shippingCost ?? this.shippingCost,
      status: status ?? this.status,
      paymentStatus: paymentStatus ?? this.paymentStatus,
      shippingAddress: shippingAddress ?? this.shippingAddress,
      phone: phone ?? this.phone,
      transactionId: transactionId ?? this.transactionId,
      items: items ?? this.items,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
      user: user ?? this.user,
    );
  }

  @override
  String toString() {
    return 'Order(id: $id, orderNumber: $orderNumber, totalAmount: $totalAmount, status: $status)';
  }

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    
    return other is Order &&
      other.id == id &&
      other.orderNumber == orderNumber;
  }

  @override
  int get hashCode => id.hashCode ^ orderNumber.hashCode;
} 

File: backend/app/Models/Order.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Order extends Model
{
    use HasFactory;
    
    protected $fillable = [
        'user_id',
        'order_number',
        'total_amount',
        'shipping_cost',
        'status',
        'payment_status',
        'shipping_address',
        'phone',
        'transaction_id'
    ];
    
    public function user()
    {
        return $this->belongsTo(User::class);
    }
    
    public function items()
    {
        return $this->hasMany(OrderItem::class);
    }
    
    public function payment()
    {
        return $this->hasOne(Payment::class);
    }
    
    // Generate a unique order number
    public static function generateOrderNumber()
    {
        return 'ORD-' . strtoupper(substr(uniqid(), 0, 8));
    }
}